menpo/io/input/base.py

```diff
@@ -42,30 +48,157 @@ def data_path_to(asset_filename):
     asset_path = data_dir_path() / asset_filename
     if not asset_path.is_file():
         raise ValueError("{} is not a builtin asset: {}".format(
-            asset_filename, ls_builtin_assets()))
+            asset_filename, builtin_assets()))
     return asset_path
 
 
-def same_name(path):
+def _import_builtin_asset(data_path_to, object_types, landmark_types,
+                          asset_name, **kwargs):
+    r"""Single builtin asset (landmark or image) importer.
+
+    Imports the relevant builtin asset from the ``./data`` directory that
+    ships with the project.
+
+    Parameters
+    ----------
+    asset_name : `str`
+        The filename of a builtin asset (see :map:`ls_builtin_assets`
+        for allowed values)
+
+    Returns
+    -------
+    asset :
+        An instantiated :map:`Image` or :map:`LandmarkGroup` asset.
+    """
+    asset_path = data_path_to(asset_name)
+    # Import could be either an image or a set of landmarks, so we try
+    # importing them both separately.
+    try:
+        return _import(asset_path, object_types,
+                       landmark_ext_map=landmark_types,
+                       landmark_attach_func=_import_object_attach_landmarks,
+                       importer_kwargs=kwargs)
+    except ValueError:
+        return _import(asset_path, landmark_types,
+                       importer_kwargs=kwargs)
+
+
+def _ls_builtin_assets(data_dir_path):
+    r"""List all the builtin asset examples provided.
+
+    Returns
+    -------
+    file_paths : list of `str`
+        Filenames of all assets in the data directory shipped with the
+        project.
+    """
+    return [p.name for p in data_dir_path().glob('*') if not p.is_dir()]
+
+
+def _register_importer(ext_map, extension, callable):
     r"""
-    Menpo's default image landmark resolver. Returns all landmarks found to have
+    Register a new importer for the given extension. This makes it much
+    simpler to add custom importers to the IO package.
```

Comment:
This docstring might read weird in a year (future menpo users won't know it was ever harder ;)

I'd drop the second sentence, and add an Example section if poss?